core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java

```diff
@@ -60,12 +62,15 @@ public SelectStatementHandler(DynamoDbClient client, TableMetadataManager metada
   public Scanner handle(Selection selection) throws ExecutionException {
     TableMetadata tableMetadata = metadataManager.getTableMetadata(selection);
     try {
-      if (ScalarDbUtils.isSecondaryIndexSpecified(selection, tableMetadata)) {
+      if (!(selection instanceof ScanAll)
+          && ScalarDbUtils.isSecondaryIndexSpecified(selection, tableMetadata)) {
         return executeQueryWithIndex(selection, tableMetadata);
       }
 
       if (selection instanceof Get) {
         return executeGet((Get) selection, tableMetadata);
+      } else if (selection instanceof ScanAll) {
+        return executeScan((ScanAll) selection, tableMetadata);
```

Comment:
I think the naming is getting confusing between `scan` and `query`.
Maybe, we can name it `executeFullScan`, rename `executeQuery` with `executeScan`.

Reply:
These `executeXXX` methods are named after the corresponding Dynamo operation that will be executed. 
This explains why it is confusing from the Scalar DB perspective. 

Scalar DB get -> Dynamo get
Scalar DB scan -> Dynamo query
Scalar DB scanAll -> Dynamo scan

I guess renaming them as you suggest would be easier to understand.

Reply:
I renamed the methods as suggested.